File: archives/python/5800.py
import sys
input = sys.stdin.readline

T = int(input())
for i in range(1, T+1):
    nums = sorted(list(map(int, input().split()))[1:])
    max_num, min_num = nums[-1], nums[0]
    largest_gap = max([nums[i+1]-nums[i] for i in range(len(nums)-1)])
    
    print(f"Class {i}")
    print(f"Max {max_num}, Min {min_num}, Largest gap {largest_gap}")

File: archives/cpp/5800.cpp
#include <bits/stdc++.h>
#define FOR(i, start, end) for (int i = start; i < end; i++)
#define endl "\n"
using namespace std;

vector<int> nums;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    cout.tie(NULL);

    int K; cin >> K;
    int cnt = 1;
    while (K--) {
        nums.clear();
        int N; cin >> N;
        int temp;
        FOR(i, 0, N) {
            cin >> temp;
            nums.push_back(temp);
        }
        sort(nums.begin(), nums.end());
        int gap = 0;
        FOR(i, 0, N-1){
            gap = max(gap, nums[i+1] - nums[i]);
        }

        int max = *max_element(nums.begin(), nums.end());
        int min = *min_element(nums.begin(), nums.end());
        cout << "Class " << cnt << endl;
        cout << "Max " << max << ", Min " << min <<", Largest gap " << gap << endl;
        cnt++;
    }

    return 0;
}